### PR TITLE
Provide a way to disable SpringManagedContext

### PR DESCRIPTION
**Pull request by emrahkocaman**
`spring-aware` config element added to `hazelcast-spring-3.5.xsd`.

By default hz uses `SpringManagedContext` to scan `SpringAware` annotations. This may cause some performance penalty even if users don't use `SpringAware`. 
Starting with 3.5 `SpringManagedContext`will be disabled by default. To enable it `<hz:spring-aware />` tag is needed to be added to the config.

### CODE REVIEW

**Comment by mesutcelik** [comment]
Can you also update document for the new element?

File: hazelcast-spring/src/test/resources/com/hazelcast/spring/springaware/springAware-disabled-applicationContext-hazelcast.xml [hunk 1]
```diff
@@ -0,0 +1,63 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ Copyright (c) 2008-2013, Hazelcast, Inc. All Rights Reserved.
```
Comment (mesutcelik) [inline hazelcast-spring/src/test/resources/com/hazelcast/spring/springaware/springAware-disabled-applicationContext-hazelcast.xml]: can you update your header text inside the intellij? if that is put by your IDE

File: hazelcast-spring/src/test/java/com/hazelcast/spring/springaware/TestDisabledSpringAwareAnnotation.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        Hazelcast.shutdownAll();
+    }
+
+    @Autowired
+    private ApplicationContext context;
+
+    @Test
+    public void test() {
+        HazelcastInstance instance = (HazelcastInstance) context.getBean("instance");
+        assertEquals(null, instance.getConfig().getManagedContext());
```
Comment (mesutcelik) [inline hazelcast-spring/src/test/java/com/hazelcast/spring/springaware/TestDisabledSpringAwareAnnotation.java]: assertNull can be used here...

**Comment by emrahkocaman** [comment]
@mesutcelik Sure, I've updated documentation and also changed default behaviour of SpringManagedContext to disabled.

File: hazelcast-spring/src/test/resources/com/hazelcast/spring/springaware/springAware-disabled-applicationContext-hazelcast.xml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+       xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+       xmlns:hz="http://www.hazelcast.com/schema/spring"
+       xsi:schemaLocation="http://www.springframework.org/schema/beans
+		http://www.springframework.org/schema/beans/spring-beans-3.0.xsd
+		http://www.hazelcast.com/schema/spring
+		http://www.hazelcast.com/schema/spring/hazelcast-spring.xsd">
+
+    <hz:hazelcast id="instance" lazy-init="true" scope="singleton">
+        <hz:config>
+            <hz:annotation-driven enabled="false"/>
```
Comment (mesutcelik) [inline hazelcast-spring/src/test/resources/com/hazelcast/spring/springaware/springAware-disabled-applicationContext-hazelcast.xml]: I somehow did not like *annotation-driven* as a name. We can add a new attribute to <hz:hazelcast> tag like <hz:hazelcast spring-aware-enabled="true">

Comment (emrahkocaman) [inline hazelcast-spring/src/test/resources/com/hazelcast/spring/springaware/springAware-disabled-applicationContext-hazelcast.xml]: Changed its name *spring-aware*.

**Comment by emrahkocaman** [comment]
Closing this PR, see https://github.com/hazelcast/hazelcast/pull/4711 instead.

**Pull request closed without merging**